pkg/skaffold/build/scheduler.go

```diff
@@ -87,19 +74,29 @@ func InParallel(ctx context.Context, out io.Writer, tags tag.ImageTags, artifact
 	return collectResults(out, artifacts, results, outputs)
 }
 
-func runBuild(ctx context.Context, cw io.WriteCloser, tags tag.ImageTags, artifact *latest.Artifact, results *sync.Map, build ArtifactBuilder) {
-	event.BuildInProgress(artifact.ImageName)
+func runBuild(ctx context.Context, cw io.WriteCloser, tags tag.ImageTags, a *artifactChanModel, results *sync.Map, build ArtifactBuilder) {
+	defer cw.Close()
+	err := a.waitForDependencies(ctx)
+	event.BuildInProgress(a.artifact.ImageName)
+	if err != nil {
+		event.BuildFailed(a.artifact.ImageName, err)
+		results.Store(a.artifact.ImageName, err)
```

Comment:
These should use `errors.Errorf("failure upstream: %w", err)` ?  Otherwise this artifact is reporting the dependency's error as its own?

Reply:
the error is already formatted as `fmt.Errorf("failed to build required artifact: %q", depStatus.imageName)`. It's indexed on the artifact name so the build failure logged is `couldn't build "artifact1": failed to build required artifact: "artifact2"`

Reply:
The event is
```json
{
  "artifact": "artifact1",
  "status": "Failed",
  "err": "failed to build required artifact: \"artifact2\""
}
```

Reply:
I think this is sufficient.

Reply:
👍  I had meant to delete this comment.